torch/distributions/distribution.py

```diff
@@ -28,9 +29,12 @@ def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size(), validate_
             self._validate_args = validate_args
         if self._validate_args:
             for param, constraint in self.arg_constraints.items():
-                if not constraints.is_dependent(constraint):
-                    if not constraint.check(self.__getattribute__(param)).all():
-                        raise ValueError("The parameter {} has invalid values".format(param))
+                if constraints.is_dependent(constraint):
+                    continue  # skip constraints that cannot be checked
+                if param not in self.__dict__ and isinstance(getattr(type(self), param, None), lazy_property):
```

Comment:
We should probably raise an error if it's not in the dict, and not in the type (I'm talking about the `None` default), because it means there's an error in `arg_constraints`